analysis/analysis-api-fir/src/org/jetbrains/kotlin/analysis/api/fir/types/KtFirFunctionalType.kt

```diff
@@ -57,8 +66,8 @@ internal class KtFirFunctionalType(
 
     override val arity: Int
         get() = withValidityAssertion {
-            if (coneType.isExtensionFunctionType) coneType.typeArguments.size - 2
-            else coneType.typeArguments.size - 1
+            if (coneType.isExtensionFunctionType) expandedType.typeArguments.size - 2
+            else expandedType.typeArguments.size - 1
```

Comment:
because it requires to modify `ownTypeArguments`? If so, we're back to this discussion: https://github.com/JetBrains/kotlin/pull/5089#discussion_r1095536135

Reply:
Done by replacing most occurrences of `coneType` with `expandedType` except:
1. `classId`, `classSymbol`, `nullability` and `asStringForDebugging`: should it be expanded or original? I'm currently following `KtFirUsualClassType`, which uses the original.
2. `isReflectType`: `functionTypeKind` will expand it.